examples/snmp/pfsense.js

```diff
@@ -0,0 +1,889 @@
+		}],
+		snmpget: false
+	},
+	{
+		label: 'System contact details',
+		unit: '',
+		oid: '1.3.6.1.2.1.1.4.0'
+	},
+	{
+		label: 'System description',
+		unit: '',
+		oid: '1.3.6.1.2.1.1.1.0'
+	},
+	{
+		label: 'System location',
+		unit: '',
+		oid: '1.3.6.1.2.1.1.6.0'
+	},
+	{
+		label: 'System name',
+		unit: '',
+		oid: '1.3.6.1.2.1.1.5.0'
+	},
+	{
+		label: 'System object ID',
+		unit: '',
+		oid: '1.3.6.1.2.1.1.2.0'
+	},
+	{
+		'label': 'Uptime',
+		'unit': 'uptime',
+		'oid': '1.3.6.1.2.1.25.1.1.0'
+	},
+];
+
+function execute_all(arrayFn, callback) {
+	if (arrayFn.length == 0) {
+		callback([]);
+	}
+	var length = arrayFn.length;
+	var results = new Array(length);
+	var finished = 0;
+	arrayFn.forEach(function (fn, index) {
+		fn(function (result) {
+			results[index] = result;
+			if (++finished == length) {
+				callback(results);
+			}
+		});
+	});
+}
+
+function execute_seq(functions, callback) {
+	var _this = this;
+	var callbackResult = null;
+	function executeNext(functionIndex) {
+		if (functionIndex == functions.length) return callback.apply(_this, [callbackResult]);
+		functions[functionIndex].apply(_this, [callbackResult, function (result) {
+			callbackResult = result;
+			executeNext.apply(_this, [++functionIndex]);
+		}]);
+	}
+	executeNext.apply(_this, [0]);
+}
+
+function get_result(oid) {
+	return snmp_parameters.filter(function (p) {
+		return p.oid == oid;
+	}).map(function (p) {
+		return p.result;
+	})[0];
+}
+
+function update_oid(param, res) {
+	for (var key in res) {
+		if (res[key] == param.walk.name) {
+			var oidArray = key.split('.');
+			var index = oidArray[oidArray.length - 1];
+			param.oid = param.walk.oid + '.' + index;
+			break;
+		}
+	}
+}
+
+function change_per_second(last, next) {
+	var old_time = this.first_execution_date;
+	var old_value = this.result;
+	var _this = this;
+	setTimeout(function () {
+		var new_time = new Date();
+		createSNMPSession().get([_this.oid], function (result, error) {
+			if (error) {
+				console.error(error);
+				D.failure(D.errorType.GENERIC_ERROR);
+			}
+			var new_value = result[_this.oid];
+			var time_diff = (new_time - old_time) / 1000;
+			_this.result = (new_value - old_value) / time_diff;
+			next(_this);
+		});
+	}, 5000);
+}
+
+function divide_by_cpu_count(last, next) {
+	var nCPU = get_result('1.3.6.1.2.1.25.3.3.1.2.cpus');
+	this.result = this.result / nCPU;
+	next(this);
+}
+
+function multiply_result(number) {
+	return function (last, next) {
+		this.result = this.result * number;
+		next(this);
+	};
+}
+
+
+
+
+function preprocess_params(cb) {
+	var arrayFn = [];
+	snmp_parameters.forEach(function (p) {
+		if (p.walk) {
+			arrayFn.push(function (callback) {
+				createSNMPSession().walk(p.walk.ref, function (result, error) {
+					p.walk.callback(p, result);
+					callback(p);
+				});
+			});
+
+		}
+	});
+
+	execute_all(arrayFn, function (result) {
+		cb();
+	});
+}
+
+
+function generate_variable(p, result, first_execution_date, callback) {
```

Comment:
Let me know once you push the table variables support so that I can review the output